TASVideos.Common/HtmlWriter.cs

```diff
@@ -4,6 +4,49 @@ namespace TASVideos.Common;
 
 public partial class HtmlWriter(TextWriter w)
 {
+	private struct HtmlClassList()
+	{
+		private List<string>? _list = null;
+
+		private string? _single = null;
+
+		public void Add(string item)
+		{
+			if (_single is not null)
+			{
+				_list = new() { _single, item };
+				_single = null;
+			}
+			else if (_list is not null)
+			{
+				_list.Add(item);
+			}
+			else
+			{
+				_single = item;
+			}
+		}
+
+		public readonly string Serialize()
+		{
+			HashSet<string> classes;
+			if (!string.IsNullOrEmpty(_single))
+			{
+				classes = _single.Split(' ', StringSplitOptions.RemoveEmptyEntries).ToHashSet();
+			}
+			else if (_list is not null)
+			{
+				classes = _list.SelectMany(static s => s.Split(' ', StringSplitOptions.RemoveEmptyEntries)).ToHashSet();
+			}
+			else
+			{
+				return "";
+			}
+
+			return classes.Count is 0 ? "" : $" class=\"{string.Join(' ', classes)}\"";
```

Comment:
Two problems here:

* We're unnecessarily allocating an additional string and making an extra copy; why not just write to the output?
* Does this handle quote and ampersand escaping?  It doesn't seem to.  Ideally, it would use the same logic as the rest of the system uses to handle that.

Reply:
Whoops! You're totally right.